src/web/rustdoc.rs

```diff
@@ -336,9 +332,10 @@ impl RustdocPage {
             result => result.context("error rewriting HTML")?,
         };
 
+        let robots = if is_latest_url { "" } else { "noindex" };
         Ok((
             StatusCode::OK,
-            TypedHeader(canonical_url),
+            AppendHeaders([("X-Robots-Tag", robots)]),
```

Comment:
I'm not 100% certain, but I believe we don't need the `AppendHeaders` here, but can just use the plain array. 

Similar to the cache `Extension` below we could then have an empty array when we want to omit the header.

Reply:
Ah, you're correct that we don't need `AppendHeaders` but can use the plain array.

Unfortunately we can't do an `if` substituting in an empty array, because `[T; 0]` and `[T; 1]` are different types. :(

Reply:
You can probably use an `Option` via <https://docs.rs/axum/latest/axum/response/trait.IntoResponseParts.html#impl-IntoResponseParts-for-Option%3CT%3E>

Reply:
Great tip, thanks!